pinot-core/src/main/java/com/linkedin/pinot/core/util/DataTableCustomSerDe.java

```diff
@@ -0,0 +1,417 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.util;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLog;
+import com.linkedin.pinot.common.utils.primitive.MutableLongValue;
+import com.linkedin.pinot.common.utils.DataTableJavaSerDe;
+import com.linkedin.pinot.core.query.aggregation.function.AvgAggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.function.MinMaxRangeAggregationFunction;
+import com.linkedin.pinot.core.query.aggregation.function.quantile.digest.QuantileDigest;
+import com.linkedin.pinot.core.segment.creator.impl.V1Constants;
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class DataTableCustomSerDe extends DataTableJavaSerDe {
```

Comment:
Let's talk about this. The explanation in DataTableBuilder does not seem
correct. We are using non-primitive types here that DataTableBuilder claims
to avoid.
If we are optimizing for speed(or size) we should have a performance
comparison to establish the fact.
-Adwait

On Mon, Sep 12, 2016 at 11:02 PM, Mayank Shrivastava <
notifications@github.com> wrote:

> In pinot-core/src/main/java/com/linkedin/pinot/core/util/
> DataTableCustomSerDe.java
> <https://github.com/linkedin/pinot/pull/532#discussion_r78500622>:
>
> > +import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
> > +import java.io.ByteArrayInputStream;
> > +import java.io.ByteArrayOutputStream;
> > +import java.io.DataInputStream;
> > +import java.io.DataOutputStream;
> > +import java.io.IOException;
> > +import java.io.Serializable;
> > +import java.io.UnsupportedEncodingException;
> > +import java.nio.ByteBuffer;
> > +import java.util.HashMap;
> > +import java.util.Map;
> > +import org.slf4j.Logger;
> > +import org.slf4j.LoggerFactory;
> > +
> > +
> > +public class DataTableCustomSerDe extends DataTableJavaSerDe {
>
> For performance reasons. Refer to the [pre-existing] explanation in the
> DataTableBuilder class.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/linkedin/pinot/pull/532/files/8c4d4200a74883aad3820b2bfe05b44f5318d12a#r78500622>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABAmRoHUAR7HyH7gOaCYhXuqHUtzz_Iuks5qpjyAgaJpZM4J7Nu8>
> .
>